WordPress/gutenberg
# FIX: Replace the AM/PM toggle ButtonGroup with ToggleGroupControl.

## What?
Fixes https://github.com/WordPress/gutenberg/issues/61163

## Why?
When the is12Hour={true} setting is enabled, the AM/PM toggle relies solely on visual cues to convey its state, which presents a challenge for screen readers. This design overlooks accessibility needs, potentially creating barriers for users who rely on such assistive technologies.

## How?
The previous ButtonGroup for toggling between AM/PM is replaced with ToggleGroupControl for better accessibility.

## Testing Instructions 
- Create a new post and add the Date block.
- Make sure your site is configured for 12 Hrs Time Format from the Settings >> General
- Now edit the time on the date block using the "Edit" option (pencil icon) in the Block Controls Toolbar.
- Test using the screen reader to read aloud the intended use of ToggleGroupControl.

### Testing Instructions for Keyboard
- Create a new post and add the Date block.
- Make sure your site is configured for 12 Hrs Time Format from the Settings >> General
- Now edit the time on the date block using the "Edit" option (pencil icon) in the Block Controls Toolbar.
- Navigate to AM/PM selector using `Tab` key.
- Once on AM/PM selector you can toggle between the two states using arrow keys.
- To select the highlighted option press `Spacebar`.

## Screenshots or screencast
<img width="429" alt="Screenshot 2024-05-10 at 12 50 35 PM" src="https://github.com/WordPress/gutenberg/assets/65603015/e3b30a41-b403-4642-8123-055420733b58">
